weave-js/src/components/Panel2/PanelFilterEditor.tsx

```diff
@@ -221,6 +222,19 @@ const SingleFilterVisualEditor: React.FC<{
   const curClause: VisualClauseWorkingState = {key, simpleKeyType, op, value};
   const valid = visualClauseIsValid(curClause);
 
+  React.useEffect(() => {
```

Comment:
Mirroring the state out with a useEffect will cause some extra react churn.

It'll be like
- setKey
- rerender
- after render this useEffect will run
- setTempClause
- rerender

So we get two renders for a single user action.

It'd be better to move the state out of SingleFilterVisualEditor entirely

Reply:
Sounds good, will move the full state out